modules/management-console-agent/src/test/java/org/apache/ignite/internal/agent/action/controller/QueryActionsControllerTest.java

```diff
@@ -544,6 +544,93 @@ public void shouldReturnRunningQueriesFromCoordinatorNode() {
         });
     }
 
+    /**
+     * 1. Create and fill the cache with Integer key and String value.
+     * 2. Execute sql query: SELECT count(*), sleep() FROM "TestCache".STRING and wait completion.
+     * 3. Execute scan query for TestCache and wait completion.
+     * 4. Send the "QueryActions.history" action with "since" argument equals to 1.
+     * 5. Assert that query history contains 2 queries.
+     * 6. Assert that SCAN and SQL_FIELDS query present in query history.
+     * 7. Assert that history queries contains a correct queries text presentation.
+     */
+    @Test
+    public void shouldReturnQueryHistory() {
```

Comment:
Please add a test for queries sent not from the GGCC API, but to Ignite directly.

Reply:
Added